frame/bags-list/src/mock.rs

```diff
@@ -0,0 +1,170 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Mock runtime for pallet-bags-lists tests.
+
+use super::*;
+use frame_election_provider_support::VoteWeight;
+use frame_support::parameter_types;
+
+pub type AccountId = u32;
+pub type Balance = u32;
+
+parameter_types! {
+	pub static NextVoteWeight: VoteWeight = 0;
+}
+
+pub struct StakingMock;
+impl frame_election_provider_support::VoteWeightProvider<AccountId> for StakingMock {
+	fn vote_weight(id: &AccountId) -> VoteWeight {
+		match id {
+			710 => 15,
+			711 => 16,
+			712 => 2_000, // special cases used for migrate test
+			_ => NextVoteWeight::get(),
+		}
+	}
+	#[cfg(any(feature = "runtime-benchmarks", test))]
+	fn set_vote_weight_of(_: &AccountId, weight: VoteWeight) {
+		// we don't really keep a mapping, just set weight for everyone.
+		NextVoteWeight::set(weight)
+	}
+}
+
+impl frame_system::Config for Runtime {
+	type SS58Prefix = ();
+	type BaseCallFilter = frame_support::traits::Everything;
+	type Origin = Origin;
+	type Index = u64;
+	type BlockNumber = u64;
+	type Call = Call;
+	type Hash = sp_core::H256;
+	type Hashing = sp_runtime::traits::BlakeTwo256;
+	type AccountId = AccountId;
+	type Lookup = sp_runtime::traits::IdentityLookup<Self::AccountId>;
+	type Header = sp_runtime::testing::Header;
+	type Event = Event;
+	type BlockHashCount = ();
+	type DbWeight = ();
+	type BlockLength = ();
+	type BlockWeights = ();
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = pallet_balances::AccountData<Balance>;
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type OnSetCode = ();
+}
+
+parameter_types! {
+	pub static BagThresholds: &'static [VoteWeight] = &[10, 20, 30, 40, 50, 60, 1_000, 2_000, 10_000];
+}
+
+impl crate::Config for Runtime {
+	type Event = Event;
+	type WeightInfo = ();
+	type BagThresholds = BagThresholds;
+	type VoteWeightProvider = StakingMock;
+}
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Runtime>;
+type Block = frame_system::mocking::MockBlock<Runtime>;
+frame_support::construct_runtime!(
+	pub enum Runtime where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic,
+	{
+		System: frame_system::{Pallet, Call, Storage, Event<T>, Config},
+		BagsList: crate::{Pallet, Call, Storage, Event<T>},
```

Comment:
```suggestion
		BagsList: bags_list::{Pallet, Call, Storage, Event<T>},
```

Reply:
and then add `use crate::{self as bags_list}`. It just helps with consistency.

Reply:
actually I find the usage of `crate` here good, we just need to provide the path to the pallet. `crate` is a valid path.